apps/frontend/src/components/fap/Proposals/AssignProposalsToFaps.tsx

```diff
@@ -0,0 +1,161 @@
+        .toString()
+  );
+
+  const initialValues: Record<string, number | undefined | null> = {};
+
+  proposalsUniqueInstruments.forEach((instrument) => {
+    initialValues[`selectedFapIds_${instrument.id}`] =
+      proposalFapInstruments?.find((i) => i?.instrumentId === instrument.id)
+        ?.fapId || null;
+  });
+
+  const hasEmptyValue = (values: Record<string, number | undefined | null>) =>
+    Object.values(values).every((v) => v !== undefined && v !== null);
+
+  return (
+    <Container
+      component="main"
+      maxWidth="xs"
+      data-cy="proposals-fap-assignment"
+    >
+      <Formik
+        initialValues={{ ...initialValues }}
+        onSubmit={async (values): Promise<void> => {
+          const fapInstruments: FapInstrumentInput[] =
+            proposalsUniqueInstruments?.map((instrument) => ({
+              instrumentId: instrument.id,
+              fapId: values[`selectedFapIds_${instrument.id}`] || null,
+            })) || [];
+
+          await assignProposalsToFaps(fapInstruments);
+          close();
+        }}
+      >
+        {({ isSubmitting, values }): JSX.Element => (
+          <Form>
+            <Typography variant="h6" component="h1">
+              Assign proposal/s to FAP
+            </Typography>
+
+            <Grid container spacing={3}>
+              <Grid item xs={12}>
+                {proposalsUniqueInstruments.map((instrument) => (
+                  <FormikUIAutocomplete
+                    key={instrument.id}
+                    name={`selectedFapIds_${instrument.id}`}
+                    label="Select FAP"
+                    loading={loadingFaps}
+                    items={faps.map((fap) => ({
+                      value: fap.id,
+                      text: fap.code,
+                    }))}
+                    InputProps={{
+                      startAdornment: (
+                        <InputAdornment position="start">
+                          {instrument?.name}:
+                        </InputAdornment>
+                      ),
+                    }}
+                    disabled={isSubmitting}
+                    noOptionsText="No FAPs"
+                    data-cy="fap-selection"
+                  />
+                ))}
+              </Grid>
+            </Grid>
+
+            {!proposalsUniqueInstruments?.length && (
+              <Alert severity="warning" data-cy="different-instruments">
+                Selected proposal/s do NOT have any instruments assigned. Please
+                assign instruments first and then you can select FAPs
+                accordingly.
+              </Alert>
+            )}
+
+            {proposalFapInstruments?.[0] && !hasEmptyValue(values) && (
+              <Alert severity="warning" data-cy="remove-fap-alert">
+                Be aware that leaving FAP selection empty will remove assigned
+                FAP from proposal/s and delete all FAP reviews on that
+                assignment.
```

Comment:
I don't like the sound of this. I don't think a user should be able to mass delete reviews after the FAP's have happened.

Reply:
Yes I undertand you. I presented this to Carina and Emma with a demo and they were fine with it for now. Maybe this will need some modifications in the future but for the initial phase it works. This action is allowed only for user officers and I have few tickets to improve their awareness that they are working with multi-instrument and FAP proposals.